spaceshooter/spaceShooter.py

```diff
@@ -435,126 +435,150 @@ def update(self):
+        hits = pygame.sprite.spritecollide(
+            player, mobs, True, pygame.sprite.collide_circle)
+        for hit in hits:
+            player.shield -= hit.radius * 2
+            expl = Explosion(hit.rect.center, 'sm')
+            all_sprites.add(expl)
+            newmob()
+            if player.shield <= 0:
+                death_explosion = Explosion(player.rect.center, 'player')
+                all_sprites.add(death_explosion)
+                player.hide()
+                player.lives -= 1
                 player.shield = 100
-        if hit.type == 'gun':
-            player.powerup()
 
-    ## if player died and the explosion has finished, end game
-    if player.lives == 0 and not death_explosion.alive():
-        running = False
-        # menu_display = True
-        # pygame.display.update()
+        # if the player hit a power up
+        hits = pygame.sprite.spritecollide(player, powerups, True)
+        for hit in hits:
+            if hit.type == 'shield':
+                player.shield += random.randrange(10, 30)
+                if player.shield >= 100:
+                    player.shield = 100
+            if hit.type == 'gun':
+                player.powerup()
+
+        if player.lives == 0 and not death_explosion.alive():
+            # running = False
+            # menu_display = True
+            screen.fill(BLACK)
+            screen.blit(background, background_rect)
+            pygame.display.update()
+            while True:
+                ev = pygame.event.poll()
+                if ev.type == pygame.KEYDOWN:
+                    if ev.key == pygame.K_RETURN:
+                        break
+                    elif ev.key == pygame.K_ESCAPE:
+                        break
+                    elif ev.key == pygame.K_r:
+                        reset = True
+                        player.lives = 5
+                        player.shield = 100
+                        score = 0
+                        break
+                elif ev.type == pygame.QUIT:
+                    break
+                else:
+
+                    draw_text(screen, "GAME OVER!", 80,
+                              WIDTH/2, (HEIGHT/2)-110)
+                    if(score >= highscore):
+                        highscore = score
+                        draw_text(
+                            screen, f"New High Score - {score}", 50, WIDTH/2, (HEIGHT/2)-30)
+                    else:
+                        draw_text(
+                            screen, f"Your Score is {score}", 50, WIDTH/2, (HEIGHT/2)-30)
+                    draw_text(screen, "Press R to Restart",
+                              30, WIDTH/2, HEIGHT/2+40)
+                    draw_text(screen, "or ESC to Exit",
+                              30, WIDTH/2, (HEIGHT/2)+70)
+                    pygame.display.update()
+            if reset == False:
+                break
+            else:
+                reset = False
+                continue
```

Comment:
You are using `reset` to quit "main" loop controlled by variable `running`, why not just use it instead?